File: program/program/cylindrically-rotate-a-matrix-by-90-degrees/cylindrically_rotate_a_matrix_by_90_degrees.pl
use strict;
use warnings;

sub cylindrical_rotate_90 {
    my ($matrix) = @_;

    my $num_rows = scalar @$matrix;
    my $num_cols = scalar @{$matrix->[0]};

    my @rotated_matrix;
    for my $i (0 .. $num_rows - 1) {
        for my $j (0 .. $num_cols - 1) {
            $rotated_matrix[$j][$num_rows - 1 - $i] = $matrix->[$i][$j];
        }
    }

    return \@rotated_matrix;
}

# Input matrix
my @input_matrix = (
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
);

# Perform cylindrical rotation
my $rotated_matrix_ref = cylindrical_rotate_90(\@input_matrix);

# Output the rotated matrix
for my $row (@$rotated_matrix_ref) {
    print join(" ", @$row), "\n";
}

File: program/program/cylindrically-rotate-a-matrix-by-90-degrees/CylindricallyRotateAMatrixBy90Degrees.swift
let input = [
	[1, 2, 3],
	[4, 5, 6],
	[7, 8, 9]
]

func rotate90Clockwise(_ matrix: [[Int]]) -> [[Int]] {
	let N = matrix.count
	var rotated = Array(repeating: Array(repeating: 0, count: N), count: N)
	for r in 0 ..< N {
		for c in 0 ..< N {
			rotated[c][N - (r + 1)] = matrix[r][c]
		}
	}
	return rotated
}

func printMatrix(_ matrix: [[Int]]) {
	for r in 0 ..< matrix.count {
		for c in 0 ..< matrix.count {
			print(matrix[r][c], terminator: " ")
		}
		print()
	}
	print()
}

print("Input  :")
printMatrix(input)
print("Output  :")
printMatrix(rotate90Clockwise(input))